command/agent.go

```diff
@@ -211,6 +212,12 @@ func (c *AgentCommand) Run(args []string) int {
 		return 1
 	}
 	c.logger = l
+	// Once we've assigned the logger, we set output so that libraries like
+	// consul-template will honor our settings (i.e. if the -log-format is JSON
+	// we want JSON, not a mix of JSON and non-JSON messages).
+	log.SetOutput(l.StandardWriter(&hclog.StandardLoggerOptions{
+		InferLevelsWithTimestamp: true,
+	}))
```

Comment:
This the the actual change, the rest of the changes in *this* file are renaming/refactoring ones.